lib/service/service.go

```diff
@@ -5665,3 +5669,10 @@ func copyAndConfigureTLS(config *tls.Config, log logrus.FieldLogger, accessPoint
 
 	return tlsConfig
 }
+
+func makeXForwardedForMiddleware(cfg *servicecfg.Config) utils.HTTPMiddleware {
+	if cfg.Proxy.UseXForwardedFor {
```

Comment:
Just thinking, should we be removing this header for a good measure if it's not enabled? 🤔

Reply:
Good question. I think we should keep it backwards compatible when this flag is not enabled by default.

For example, well looking at this, I found our app access doc explicitly says app header rewrite features skips `X-Forwarded-*`.  https://goteleport.com/docs/application-access/guides/connecting-apps/